### PR TITLE
Workflow Submission: Add custom name to a workflow execution

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

Users can add a custom name to a workflow during the submission process.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

**Light Mode**
image
 
**Dark Mode**
image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Open the Workflow Submission dialog
2. There should always be a field at the top for a custom name

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/components/nextflow_component.en.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+en:
+  name:
+    label: Name (Optional)
+    helper: A custom name will make it easier to search for this workflow execution in the future.
```
Comment (ericenns) [inline app/components/nextflow_component.en.yml:4]: **nitpick (blocking):** Can you change the helper text to `A custom name will it easier to search for this in the future.`

That way this also makes sense on the automated workflow executions page.

**Review by ericenns — CHANGES_REQUESTED** [review]
One minor translation comment, but also could you please add `name` to `AutomatedWorkflowExecution` model.

Are you also going to update the `WorkflowExecutions` listing to include name in the table and in the show page?

File: app/components/nextflow_component.en.yml (hunk 1 above)
Comment (joshsadam) [reply app/components/nextflow_component.en.yml]: Updated in b01e147

File: app/models/workflow_execution.rb [hunk 2]
```diff
@@ -63,6 +63,7 @@ def sent_to_ga4gh?
 
   def as_wes_params
     {
+      name:,
```
Comment (ericenns) [inline app/models/workflow_execution.rb:66]: **nitpick (blocking):** this needs to be removed `name` is not a valid wes param.

File: app/views/projects/automated_workflow_executions/_automated_workflow_execution.html.erb [hunk 3]
```diff
@@ -5,6 +5,18 @@
   <td class="px-6 py-4 whitespace-nowrap">
     <%= automated_workflow_execution.id %>
   </td>
+  <td
+    class="
+      px-6
+      py-4
+      font-medium
+      text-gray-900
```
Comment (ericenns) [inline app/views/projects/automated_workflow_executions/_automated_workflow_execution.html.erb:13]: **nitpick (blocking):** Shouldn't this be `text-slate-900`

File: app/components/nextflow_component.html.erb
Comment (ericenns) [inline app/components/nextflow_component.html.erb:1]: **nitpick (blocking):** Can you change all instance of `gray` in here to `slate`

**Review by ericenns — CHANGES_REQUESTED** [review]
Looking good, just a few comments

File: app/models/workflow_execution.rb (hunk 2 above)
Comment (joshsadam) [reply app/models/workflow_execution.rb:66]: Removed in b6ef470f60c7206f0c965493a15ec12e6cbacb24

File: app/views/projects/automated_workflow_executions/_automated_workflow_execution.html.erb (hunk 3 above)
Comment (joshsadam) [reply app/views/projects/automated_workflow_executions/_automated_workflow_execution.html.erb]: Updated in 20db5c7

File: app/components/nextflow_component.html.erb
Comment (joshsadam) [reply app/components/nextflow_component.html.erb:1]: Updated in 20db5c7

File: app/views/projects/automated_workflow_executions/_automated_workflow_execution.html.erb (hunk 3 above)
Comment (joshsadam) [reply app/views/projects/automated_workflow_executions/_automated_workflow_execution.html.erb]: Updated in 20db5c7

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**